src/Rules/RuleMetaDataProvider.cs

```diff
@@ -56,7 +59,11 @@ public async Task<IRuleInfo> GetRuleInfoAsync(SonarCompositeRuleId ruleId, Cance
 
             //TODO: this does not seem to support taint.
             var language = Language.GetLanguageFromRepositoryKey(ruleId.RepoKey);
-            Debug.Assert(language != null, $"Unable to determine the language from the repo key. RepoKey: {ruleId.RepoKey}");
+
+            // It's possible we'll be asked for help for a language we don't handle locally, in which
+            // case we'll return null.
+            // See https://github.com/SonarSource/sonarlint-visualstudio/issues/4582
+            if (language == null) { return null; }
```

Comment:
Otherwise we get a null-ref exception later when we try to use `language`.

Reply:
If null is returned the `Education` service will try to show the help on rules.sonarsource.com.
See [here](https://github.com/SonarSource/sonarlint-visualstudio/blob/7543d596f851c0d09b910da7996162d6c781531b/src/Education/Education.cs#L85).